es6.md

```diff
@@ -5,6 +5,7 @@ This document specifies the extensions to the core ESTree AST types to support t
 ```js
 extend interface Program {
     sourceType: "script" | "module";
+    body: [ Statement | ModuleDeclaration ];
```

Comment:
It can't be a separate PR because, once this is introduced, ModuleDeclarations are allowed when `sourceType` is "script". Too late to restrict it after that. If we don't care about strict backwards compatibility in the interface, though, it would be fine.